# Assignment: Extracting Static WebPage

Extract information about “วันพระ” for 3 years from:
- https://www.myhora.com/ปฏิทิน/วันพระ-พ.ศ.2565.aspx
- https://www.myhora.com/ปฏิทิน/วันพระ-พ.ศ.2566.aspx
- https://www.myhora.com/ปฏิทิน/วันพระ-พ.ศ.2567.aspx


Note that you can use dateparse package to parse Thai date.  First, we will have to install the package, this is for Google Colab users.  Otherwise, installing via command line is recommended (pip or conda).

In [1]:
import sys
IN_COLAB = 'google.colab' in sys.modules
if IN_COLAB:
    %pip install dateparser

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 295.0/295.0 kB 4.0 MB/s eta 0:00:00


In [2]:
import dateparser

To convert from thai text date string, we will use the parse method.  Note that the parse method assumes the BC year, not BE.  Thus, we will have to subtract 543 from the year.  In addition, weekday() returns day of week with 0=Monday, ..., 6=Sunday.

In [3]:
dt = dateparser.parse('วันศุกร์ที่ 17 มกราคม 2563')

# this will print out weekday == 0 (Monday)
print(dt)
print(dt.weekday())

# this will print out weekday == 4 (Friday)
dt = dt.replace(year=dt.year-543)
print(dt)
print(dt.weekday())

2563-01-17 00:00:00
0
2020-01-17 00:00:00
4


In [4]:
dt = dateparser.parse('วันเสาร์ที่ 21 กันยายน 2564')
dt = dt.replace(year=dt.year-543)
print(dt)
print(dt.weekday())

2021-09-21 00:00:00
1


Count the distribution of number of week days that are “วันพระ” for all three years and answer the following questions:

In [5]:
import requests
import re
from bs4 import BeautifulSoup

In [6]:
buddhist1 = requests.get('https://www.myhora.com/ปฏิทิน/วันพระ-พ.ศ.2565.aspx')
buddhist2 = requests.get('https://www.myhora.com/ปฏิทิน/วันพระ-พ.ศ.2566.aspx')
buddhist3 = requests.get('https://www.myhora.com/ปฏิทิน/วันพระ-พ.ศ.2567.aspx')

In [7]:
soup1 = BeautifulSoup(buddhist1.text, "lxml")
soup2 = BeautifulSoup(buddhist2.text, "lxml")
soup3 = BeautifulSoup(buddhist3.text, "lxml")

## How many วันพระ in total (of 3 years)?

In [8]:
bud_list = soup1.select('div.bud-day')+soup2.select('div.bud-day')+soup3.select('div.bud-day')
len(bud_list)

152

**ANSWER** 152 days

## How many days in total (of 3 years) that วันพระ is Monday?

In [9]:
bud_list[0].find('div', class_='bud-day-col').text

'วันอาทิตย์ที่ 2 มกราคม 2565'

In [10]:
countDay = [0]*7

for e in bud_list:
  dt = dateparser.parse(e.find('div', class_='bud-day-col').text)
  dt = dt.replace(year=dt.year-543)
  countDay[dt.weekday()] += 1

print(countDay[0])

21


**ANSWER** 21 days

In [11]:
countDay

[21, 20, 22, 23, 21, 21, 24]

## Which day of the week that has the minimum number of วันพระ?

In [12]:
print(countDay.index(min(countDay)))

1


**ANSWER** Tuesday

## Which day of the week that has the maximum number of วันพระ?

In [13]:
print(countDay.index(max(countDay)))

6


**ANSWER** Sunday